rs/nns/handlers/lifeline/impl/lifeline.mo

```diff
@@ -8,8 +8,25 @@ actor {
     private let governanceCanister : Principal = Prim.principalOfActor Governance;
     private let root : Principal = Prim.principalOfActor Root;
 
-    type UpgradeRootProposalPayload = { wasm_module : Blob; module_arg : Blob; stop_upgrade_start : Bool };
-    type HardResetRootToVersionPayload = { wasm_module : Blob; init_arg : Blob; };
+    type UpgradeRootProposalPayload = {
+      wasm_module : Blob;
+      module_arg : Blob;
+      stop_upgrade_start : Bool;
+    };
+    type HardResetRootToVersionPayload = {
+      wasm_module : Blob;
+      init_arg : Blob;
+    };
+    type CanisterIdRecord = { canister_id : Principal };
+    type CanisterSettings = {
+      controllers : ?[Principal];
+      compute_allocation: ?Nat;
+      memory_allocation: ?Nat;
+      freezing_threshold: ?Nat;
+      reserved_cycles_limit: ?Nat;
+      wasm_memory_limit: ?Nat;
+      log_visibility: ? {#controllers; #public_};
```

Comment:
Note: using `public_` instead of `public` because `public` is a keyword. I verified that if I change `public_` to something else, the state machine test fails.

Reply:
Welcome to Motoko! :-)